sdk/ml/azure-ai-ml/tests/conftest.py

```diff
@@ -538,6 +539,29 @@ def generate_compononent_hash(*args, **kwargs):
             side_effect=generate_compononent_hash
         )
 
+    if is_live():
+        # Disable on-disk cache when doing live recording as recorded requests may be impacted by
+        # the order to run tests. If we have enabled on-disk cache, tests run later will use the
+        # cached result from tests run earlier, so we won't found enough recordings when running
+        # tests in a different order.
+        if not is_live_and_not_recording():
```

Comment:
Seems this change only takes affect when someone run tests without recording locally then playback. Do we need some tests to guard it's behavior?

Reply:
1. It will still take effect within a test if multiple `jobs.create_or_update` calls are made.
2. Will add some specific tests for this.

Reply:
BTW, our daily live test for `pipeline_test` is live and not recording, right?

Reply:
Updated the fixture to avoid disable on-disk cache. Clear cache conditionally instead.

Reply:
Misunderstood your question. I think the mentioned change has been merged with below and is well tested in ci now